src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java

```diff
@@ -998,9 +1056,31 @@ public ArgumentDefinition(
             //null collections have been initialized by createCollection which is called in handleArgumentAnnotation
             //this is optional if it's specified as being optional or if there is a default value specified
             this.optional = annotation.optional() || ! this.defaultValue.equals(NULL_STRING);
+            this.maxValue = annotation.maxValue();
+            this.minValue = annotation.minValue();
+            this.maxRecommendedValue = annotation.maxRecommendedValue();
+            this.minRecommendedValue = annotation.minRecommendedValue();
+            // bounds should be only set for numeric arguments and if the type is integer it should
+            // be set to an integer
+            this.type = CommandLineParser.getUnderlyingType(this.field);
+            if (! Number.class.isAssignableFrom(this.type)) {
+                if (this.maxValue != Double.POSITIVE_INFINITY
+                        || this.minValue != Double.NEGATIVE_INFINITY
+                        || this.maxRecommendedValue != Double.POSITIVE_INFINITY
+                        || this.minRecommendedValue != Double.NEGATIVE_INFINITY) {
+                    throw new CommandLineException.CommandLineParserInternalException("Bounds for arguments could only be set for numeric arguments");
+                }
+            }
+            if (Integer.class.isAssignableFrom(this.type)) {
+                if (!isInfinityOrMathematicalInteger(this.maxValue)
+                        || !isInfinityOrMathematicalInteger(this.minValue)
+                        || !isInfinityOrMathematicalInteger(this.maxRecommendedValue)
+                        || !isInfinityOrMathematicalInteger(this.minRecommendedValue)) {
+                    throw new CommandLineException.CommandLineParserInternalException("Bounds for integer arguments should be mathematical integers");
```

Comment:
Same thing for this; test of the message should include the offending field name.

Also, please add a test for this code path.